packages/safe-tools-client/app/services/safes.ts

```diff
@@ -45,6 +46,16 @@ export default class SafesService extends Service {
 
   @tracked selectedSafe?: Safe;
 
+  constructor(properties?: object | undefined) {
+    super(properties);
+
+    // We keep reloading the token balances so that they are up do date
+    // when user adds funds to the safe independently of the app
+    setInterval(() => {
```

Comment:
Consider using a while loop with yield in an ember-concurrency task, so that the task is canceled when the service is torn town.

Reply:
I was actually considering this but does a service ever get torn down? Aren't they just singletons that live until the app is running?

Reply:
Yes, but not in tests, and that will drive you crazy with weird errors.

Reply:
I see! Ok, will adjust it.

Reply:
Added the change, please check.

Reply:
Looks good. I left one small comment. Address if you agree and good to merge.

Reply:
@jurgenwerk I saw that there were still issues and I pushed a commit to your branch that fixes them.